v1/partial/uncompressed.go

```diff
@@ -97,13 +136,13 @@ func (i *uncompressedImageExtender) Manifest() (*v1.Manifest, error) {
 		},
 	}
 
-	diffIDs, err := i.DiffIDs()
+	ls, err := i.Layers()
 	if err != nil {
 		return nil, err
 	}
 
-	for _, diffID := range diffIDs {
-		rdr, err := i.Layer(diffID)
+	for _, l := range ls {
+		rdr, err := l.Compressed()
```

Comment:
`l` here implements `Size()` and `Digest()`, but you're calling `l.Compressed()` digesting the contents manually.

Calling Size() and Digest() would be wasteful because it would digest the contents twice, but that seems like an issue with `uncompressedLayerExtender`s implementation... thoughts on memoizing that or something?